### PR TITLE
Remove KResponsiveElementMixin

### LINKED ISSUE #711: Make _useKResponsiveElement composable public and Remove KResponsiveElementMixin
## Product
KDS, Kolibri (Studio and KDP doesnt seem to be using the KResponsiveElementMixin).

## Description

Transition the internal composable [_useKResponsiveElement](https://github.com/learningequality/kolibri-design-system/blob/6a6865f1060e1da160b77b7b851f82f340f453a1/lib/composables/_useKResponsiveElement.js) to a public-facing composable named `useKResponsiveElement`. Then, remove the [KResponsiveElementMixin](https://github.com/learningequality/kolibri-design-system/blob/6a6865f1060e1da160b77b7b851f82f340f453a1/lib/KResponsiveElementMixin.js) mixin from the codebase. This involves the following steps:

* Rename `_useKResponsiveElement.js` file to `useKResponsiveElement.js`.
* Update all internal references to use the new file name.
* Migrate all internal references of `KResponsiveElementMixin` to use the new `useKResponsiveElement` composable.
* Remove the `KResponsiveElementMixin.js` file.
* Create a new documentation page for `useKResponsiveElement`. 
* Remove the old documentation page of `KResponsiveElementMixin`.

## The Value Add
After this, we will able to replace all KResponsiveElement mixins with this composable. This aligns with our plan of adopting composables over mixins.

### PR DESCRIPTION
**Pull request by AlexVelezLl**
## Description

Make useKResponsiveElement public and remove KResponsiveElementMixin.

#### Issue addressed
Addresses #711.

## Changelog

  - **Description:** Removes KResponsiveElementMixin.
  - **Products impact:** removed API.
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/711.
  - **Components:** KResponsiveElementMixin.
  - **Breaking:** yes.
  - **Impacts a11y:** no.
  - **Guidance:** Replace any use of `KResponsiveElementMixin` with the new composable `useKResponsiveElement`.

  - **Description:** Adds new useKResponsiveElement composable.
  - **Products impact:** new API.
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/711.
  - **Components:** -.
  - **Breaking:** no.
  - **Impacts a11y:** no.
  - **Guidance:** -.

  - **Description:** Remove use of KResponsiveElementMixin internally
  - **Products impact:** - none
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/711.
  - **Components:** KBreadcrumbs, KFixedGrid.
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -.
 

## Steps to test

1. Look for any reference to KResponsiveElementMixin in the lib or in the docs.
2. Do KBreadcrumbs or KFixedGrid have any regressions?

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [x] The change is described in the changelog section above

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

### CODE REVIEW

**Comment by MisRob** [comment]
Hi @AlexVelezLl, I'm getting Page Not Found when trying to access the new documentation page https://deploy-preview-783--kolibri-design-system.netlify.app/usekresponsiveelement, any idea why?

File: docs/pages/useKResponsiveElement.vue
Comment (MisRob) [inline docs/pages/useKResponsiveElement.vue:1]: There was a warning in the previous mixin's documentation page about performance issues and related discouragement of using it frequently. Is this no longer issue or do we need to move the note here?

Comment (AlexVelezLl) [reply docs/pages/useKResponsiveElement.vue:1]: Hi! It seems this is not a problem anymore, in the past we were using `ResizeSensor` which was less performant than the new `ResizeObserver`. And the `useKResponsiveElement` was optimized to use [just one `ResizeObserver` instance](https://github.com/AlexVelezLl/kolibri-design-system/blob/ba30a3e84fc6218547b7c182c0f1dc70a50627b4/lib/composables/useKResponsiveElement.js#L7) to watch all changes, which have a [good performance even with this a lot of divs](https://groups.google.com/a/chromium.org/g/blink-dev/c/z6ienONUb5A/m/F5-VcUZtBAAJ).

**Comment by AlexVelezLl** [comment]
Thank you for noticing the not found page @MisRob! It was beacuse I was using a wrong doc file name. I just updated this and its solved now!

**Review by nucleogenesis — COMMENTED** [review]
The code changes look good to me.

One thing that came to mind is that maybe we should deprecate KResponsiveElementMixin rather than just removing it outright. This would give us time to make the requisite changes in Kolibri.

Perhaps we could keep it and put a `consol.warn()` in it's `created()` hook saying "KResponsiveElement has been deprecated and will be removed in the next release. Please use `useKResponsiveElement`".

In any case, once this merged can you make an issue in Kolibri for making the necessary changes there

File: docs/pages/useKResponsiveElement.vue
Comment (MisRob) [reply docs/pages/useKResponsiveElement.vue:1]: Nice, that's good news. No note needed then.

**Comment by AlexVelezLl** [comment]
Thank you @nucleogenesis! Yes, the ideal case is deprecate it before we remove it, but we discussed this with Richard and we wanted to make the removal part of the breaking changes of KDS v5.

As per Richard's comment talking:
> Yes, I think that would be good (especially as we have already explicitly deprecated the KResponsiveWindowMixin - not sure we did an explicit deprecation warning for KResponseElementMixin, but it's used significantly less - I think only once in Kolibri).

And yes, there is already an issue in Kolibri that is blocked by this PR https://github.com/learningequality/kolibri/issues/12523, it is currently marked as help wanted, but I can take care of this and update the dependanbot PR updating the kds version as is a very straightforward change.

**Review by nucleogenesis — APPROVED** [review]
Everything looks good to me w/ that clarification @AlexVelezLl thank you!

**Pull request merged**
